### ISSUE TITLE
On Spring Boot server restart account is preserved but authority/role information is lost

### ISSUE THREAD

**Issue by george-hawkins-aa** [stormpath/stormpath-sdk-java#368]
If user's are logged in while you restart your Spring Boot web server then they are _not_ automatically logged out.

If they navigate to another page or refresh the current page then everything seems fine - the `"account"` attribute in their session survives the restart.

However the corresponding `Authentication` token is lost so any checks based around authorities or roles start failing for these users.

So after a restart user sessions end up in a kind of confusing in-between state.

I've put [a project](https://github.com/george-hawkins-aa/stormpath-spring-boot-example) up on Github that demonstrates this - see the [server restarts and authentication section](https://github.com/george-hawkins-aa/stormpath-spring-boot-example#server-restarts-and-authentication) of the README for an explanation of what's going on.

In short I think the JWT login shouldn't recover the session if the user's session has clearly been lost (as the result of a restart or whatever). And then it should be left up to the framework user to implement session persistence if needed.

**Comment by george-hawkins-aa** [comment]
OK - I've added a solution for this into my Stormpath demonstration project.

The root problem is that the existing Stormpath logic, for the classic servlet setup, is failing to take into account the additional expectations of Spring Security.

I've added a new filter (called `MissingAuthenticationFilter`) that rebuilds the Spring authentication token when necessary to bring things back into sync - preventing you from ending up in the situation where you've got a valid Stormpath `DefaultAccount` associate with your session but no corresponding authentication token (and therefore no related role/authority information).

This is discussed in more detail [in the README](https://github.com/george-hawkins-aa/stormpath-spring-boot-example#server-restarts-and-authentication). The main relevant checkin is [`c0f008f`](https://github.com/george-hawkins-aa/stormpath-spring-boot-example/commit/c0f008f).

I think this is **important**. The current situation results in user sessions ending up in a weird inconsistent state following a server restart - users remain logged in but part of their security related information is mysteriously lost.

**Comment by dogeared** [comment]
@george-hawkins-aa Thanks so much for all the legwork on this!

I've added this issue into the current milestone, `1.0.RC9`

**Comment by mzumbadoate** [comment]
After talking with @dogeared it looks like this was fixed with PR
https://github.com/stormpath/stormpath-sdk-java/pull/451
To test it I used spring-boot-default exmple
1. using the release 1.0.RC8
2. The login was successful
3. I click the restricted button and got 
"Hello, Miguel. You have the required persmissions to access this restricted resource."
4. restarted the server and when I got to localhost:8080 I got "Hello, Miguel!" this means the account was recovered.
5. when I clicked on "restricted" I got redirected to the same welcome page, this means that the Authentication token was not recovered.

Then using the current release, 1.0.RC9
I repeated the steps above but in the step 5, when I clicked "restricted" after the server restart and with the account recovered I got "Hello, Miguel. You have the required persmissions to access this restricted resource." meaning this that the Authentication token recovered.

**Comment by george-hawkins-aa** [comment]
I can confirm that the authentication token is now recreated on _every_ request so this issue no longer occurs:
```
Daemon Thread [http-nio-8080-exec-2]
    UsernamePasswordAuthenticationToken.<init>(Object, Object, Collection<GrantedAuthority>) line: 72
    UsernamePasswordAuthenticationTokenFactory.createAuthenticationToken(Object, Object, Collection<GrantedAuthority>, Account) line: 38
    StormpathAuthenticationProvider.authenticate(Authentication) line: 370
    SpringSecurityResolvedAccountFilter.filter(HttpServletRequest, HttpServletResponse, FilterChain) line: 36
    SpringSecurityResolvedAccountFilter(HttpFilter).doFilter(ServletRequest, ServletResponse, FilterChain) line: 249
    FilterChainProxy$VirtualFilterChain.doFilter(ServletRequest, ServletResponse) line: 330
    SecurityContextHolderAwareRequestFilter.doFilter(ServletRequest, ServletResponse, FilterChain) line: 169
    FilterChainProxy$VirtualFilterChain.doFilter(ServletRequest, ServletResponse) line: 330
    RequestCacheAwareFilter.doFilter(ServletRequest, ServletResponse, FilterChain) line: 48
    FilterChainProxy$VirtualFilterChain.doFilter(ServletRequest, ServletResponse) line: 330
    ...
```
